### PR TITLE
move resource to separated object in schema form

### LINKED ISSUE #1702: Extensibility: Update form widgets with external references
Update form widgets with external resources to have a cleaner API.

For example `Resource` widget has this format:
```
{
  "path": "spec.gateway",
  "widget": "Resource",
  "kind": "Gateway",
  "scope": "namespace",
  "namespace": "kyma-system",
  "group": "networking.istio.io",
  "version": "v1alpha3"
}
```
`kind`, `scope`, `namespace` and `group` should be grouped together as a single object.
Also, consider the possibility of using `dataSources` (#1700) instead

### PR DESCRIPTION
**Pull request by mrCherry97**
**Description**

Changes proposed in this pull request:

- moved resource details to separated `resource` field in schema for cleaner API
- repair `slashShortcut` in Extensions
- ...

**Related issue(s)**
Closes #1702

### CODE REVIEW

**Review by akucharska — CHANGES_REQUESTED** [review]

File: docs/extensibility/form-widgets.md [hunk 1]
```diff
@@ -115,9 +120,10 @@ ResourceRefs widgets render the lists of dropdowns to select the associated reso
 
 #### Widget-specific parameters
 
-- **kind** - _[required]_ Kubernetes kind of the resource.
-- **group** - API group used for all requests. Not provided for Kubernetes resources in the core (also called legacy) group.
-- **version** - _[required]_ API version used for all requests.
+- **resource**:
+  - **kind** - _[required]_ Kubernetes kind of the resource.
+  - **group** - API group used for all requests. Not provided for Kubernetes resources in the core (also called legacy) group.
+  - **version** - _[required]_ API version used for all requests.
```
[the remark this answers is not in the document]
Comment (mrCherry97) [reply docs/extensibility/form-widgets.md:126]: There wasn't `scope` and `namespace` before, so I think no

File: core-ui/src/components/Extensibility/components-form/ResourceRenderer.js [hunk 2]
```diff
@@ -20,11 +20,8 @@ export function ResourceRenderer({
   const { namespaceId } = useMicrofrontendContext();
 
   const { tFromStoreKeys } = useGetTranslation();
-  const group = schema.get('group');
-  const version = schema.get('version');
-  const kind = schema.get('kind');
-  const scope = schema.get('scope') || 'cluster';
-  const namespace = schema.get('namespace') || namespaceId;
+  const { group, version, kind, scope = 'cluster', namespace = namespaceId } =
+    schema.get('resource') || {};
```
[the remark this answers is not in the document]
Comment (Wawrzyn321) [reply core-ui/src/components/Extensibility/components-form/ResourceRenderer.js:24]: Nope, we don't need to do that before the `beta` tag.

**Review by akucharska — APPROVED** [review]

**Review by grego952 — APPROVED** [review]

**Pull request merged**